source/frontend/src/components/common/GenericModal.tsx

```diff
@@ -134,29 +139,65 @@ export const GenericModal = (props: Omit<BsModalProps, 'onHide'> & ModalProps) =
     }
   };
 
+  const getHeaderIcon = () => {
+    if (headerIcon) {
+      return <>{headerIcon}</>;
+    }
+
+    switch (className) {
+      case 'info':
+      case 'warning': {
+        return <FaExclamationCircle size={22} />;
+      }
+      case 'error': {
+        return <FaTimesCircle size={22} />;
+      }
+      default: {
+        return null;
+      }
+    }
+  };
+
   return (
-    <ModalContainer {...rest} show={showState} modalSize={modalSize} onHide={noop}>
+    <ModalContainer
+      {...rest}
+      show={showState}
+      modalSize={modalSize}
+      onHide={noop}
+      className={className}
+    >
       <Modal.Header closeButton={closeButton} onHide={close}>
-        <Modal.Title>{title}</Modal.Title>
+        <Modal.Title>
+          {getHeaderIcon() && <div className="header-icon">{getHeaderIcon()}</div>}
```

Comment:
nit: call this once and store the result somewhere, instead of calling this twice.

Reply:
Updated